packages/kbn-config-schema/src/types/stream_type.test.ts

```diff
@@ -56,8 +56,30 @@ test('includes namespace in failure', () => {
 
 describe('#defaultValue', () => {
   test('returns default when undefined', () => {
-    const value = new Stream();
-    expect(schema.stream({ defaultValue: value }).validate(undefined)).toStrictEqual(value);
+    // The symbol `kCapture` is set directly on the `new Stream()` object, but
+    // will be inherited by the object returned by `schema.stream()`. This means
+    // that the `toStrictEqual` check will fail because the symbol exists
+    // directly on one object but not the other. We hack this by persisting it
+    // onto the object we're testing.
+
+    const defaultValue = new Stream();
+    const value = schema.stream({ defaultValue }).validate(undefined);
+
+    // Ugly hack
+    const sym = Object.getOwnPropertySymbols(defaultValue).find(
+      key => String(key) === 'Symbol(kCapture)'
+    ) as symbol;
+
+    // Safe guards in case Node.js core or the type definitions change later
+    if (!sym) throw new Error('Could not find symbol on stream');
+    if (!(sym in value)) throw new Error('Object expected to contain symbol');
+    if (Object.hasOwnProperty.call(value, sym))
+      throw new Error('Symbol not expected to be directly on object');
+
+    // @ts-ignore
+    value[sym] = value[sym];
+
+    expect(value).toStrictEqual(defaultValue);
```

Comment:
Would this be a better solution?

```suggestion
    const value = new Stream();
    expect(schema.stream({ defaultValue: value }).validate(undefined))).toMatchInlineSnapshot(`
      Stream {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
      }
    `);
```

Reply:
I'm not sure to totally get that one. Does that mean that `schema.stream({ defaultValue: value }).validate(undefined) !== value` ?

Reply:
Yes. They are two different objects, but they are supposed to have the same structure. The name `toStrictEqual` is a bit misleading as it implies `===`, but in reality, it checks that the two objects are of the same type and have the same properties with the same values.

Reply:
I'm really surprised that the `defaultValue` reference passed to schema/joi is not the one returned when default applies, but I trust you on that one.

Reply:
I pushed a commit that changes to use the proposed `toMatchInlineSnapshot` instead as I think it's cleaner. Please let me know if I should change it back.

Reply:
Regarding if it returns the `defaultValue` or not: Below is the code for `schema.stream()`

https://github.com/elastic/kibana/blob/d1911ec843d6de590b0c24feee3942face2d4935/packages/kbn-config-schema/src/index.ts#L76-L78

As you can see it always returns a new instance of `StreamType` no matter the value of `defaultValue`. The `StreamType` object, however, is an [extends](https://github.com/elastic/kibana/blob/d1911ec843d6de590b0c24feee3942face2d4935/packages/kbn-config-schema/src/types/stream_type.ts#L25) `Type<Stream>`.

Reply:
`schema.stream` does return a new `StreamType` instance at every call.

However, `schema.stream({ defaultValue }).validate(undefined)` returns a `Stream`:

https://github.com/elastic/kibana/blob/94b2b83bd7e7f5989ae1d3ae3e977f9375403986/packages/kbn-config-schema/src/types/type.ts#L71

That's this call's return value we are asserting against in the test and the one I'm surprised about. I'm expecting it to be the same reference as the one passed as the `defaultValue` parameter when validating `undefined`.

Reply:
Ah yes, good point. Need more ☕️

However, both in `master` with Node.js 10 and in this branch with Node.js 12 they are not equal. I'm having trouble following what happens inside of `validate` as I can't seem to find where `internals.validate` is defined, so I can't check exactly what's going on.

If I console.log the two objects in Node.js 12, you can also easily see the difference:

```js
// new Stream()
Stream {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  [Symbol(kCapture)]: false
}

// validate()
Stream {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined
}
```